src/message.rs

```diff
@@ -1432,7 +1432,10 @@ pub(crate) fn rfc724_mid_exists(
     context: &Context,
     rfc724_mid: &str,
 ) -> Result<Option<(String, u32, MsgId)>, Error> {
-    ensure!(!rfc724_mid.is_empty(), "empty rfc724_mid");
+    if rfc724_mid.is_empty() {
+        warn!(context, "Empty rfc724_mid passed to rfc724_mid_exists");
+        return Ok(None);
+    }
```

Comment:
but this ensure is there since september 2019 -- why does it cause problems now? 
passing in an empty mid does seem like an error but i guess we can also just say "does not exist" anyway.

Reply:
It is not an error to pass empty message-id: it happens when there is no message-id or it cannot be parsed, empty string comes from `unwrap_or_default()` handling of message-id parsing errors.

It only happens now because of `mailparse` update that (correctly) treats Message-IDs without `<` and `>` as parsing error.

Reply:
i suggest to be careful to argue with "correctly" here -- changing behaviour of such low-level parsing functions is always dangerous and easily incorrect for users. 

anyway, IIUC we get an empty rfc724_mid because mailparse now errors on non-`<>` wrapped message ids. A bit weird but i guess it's ok to say "we don't know this rfc724_mid".

Reply:
> i suggest to be careful to argue with "correctly" here -- changing behaviour of such low-level parsing functions is always dangerous and easily incorrect for users.

Yes, it is only correct according to the RFC. But even with more robust message-ID parsing that DC had before and that @r10s is reintroducing in #1462 we can get a message without any Message-ID at all. If it reaches this check, DC is stuck in fetch loop anyway.